src/app/content/utils/archiveTreeUtils.ts

```diff
@@ -31,11 +31,34 @@ export const findDefaultBookPage = (book: {tree: ArchiveTree}) => {
   return getFirstTreeSectionOrPage(book.tree);
 };
 
+const sectionMatcher = (pageId: string) => (section: ArchiveTreeSection) =>
+    stripIdVersion(section.shortId) === stripIdVersion(pageId)
+    || stripIdVersion(section.id) === stripIdVersion(pageId);
+
 export const findArchiveTreeSection = (
   book: {tree: ArchiveTree},
   pageId: string
 ): LinkedArchiveTreeSection | undefined =>
-  flattenArchiveTree(book.tree).find((section) =>
-    stripIdVersion(section.shortId) === stripIdVersion(pageId)
-    || stripIdVersion(section.id) === stripIdVersion(pageId)
-  );
+  flattenArchiveTree(book.tree).find(sectionMatcher(pageId));
+
+interface Sections {
+  prev?: LinkedArchiveTreeSection | undefined;
+  next?: LinkedArchiveTreeSection | undefined;
+}
+
+export const prevNextBookPage = (
+  book: {tree: ArchiveTree},
+  pageId: string
+): Sections | undefined => {
+  const flattenTree = flattenArchiveTree(book.tree);
+  const index = flattenTree.findIndex(sectionMatcher(pageId));
+
+  if ( index !== -1) {
```

Comment:
you should remove this condition, then the function will always return `Sections` even if both `prev` and `next` are undefined. that will simplify the signatures for this function and the selector